pydocstringformatter/_formatting/formatters_numpydoc.py

```diff
@@ -0,0 +1,85 @@
+from __future__ import annotations
+
+from pydocstringformatter._formatting.base import NumpydocSectionFormatter
+
+
+class NumpydocSectionOrderingFormatter(NumpydocSectionFormatter):
+    """Change section order to match numpydoc guidelines."""
+
+    name = "numpydoc-section-order"
+
+    numpydoc_section_order = (
+        # I'm folding summary, deprecation warning, and extended
+        # summary into "before first heading"
+        "Parameters",
+        "Attributes",
+        "Methods",
+        "Returns",
+        "Yields",
+        "Receives",
+        "Other Parameters",
+        "Raises",
+        "Warns",
+        "Warnings",
+        "See Also",
+        "Notes",
+        "References",
+        "Examples",
+    )
+
+    def treat_sections(self, sections: list[list[str]]) -> list[list[str]]:
+        """Sort the numpydoc sections into the numpydoc order."""
+        # section_names = [sec_lines[0].strip() for sec_lines in sections]
```

Comment:
?

Reply:
For the docstring:
https://numpydoc.readthedocs.io/en/latest/format.html#sections
For the commented-out prototype code;
I probably thought I would need the section names, then realized I didn't for this formatter (there was a while when I wasn't sure if there was a way to separate numpydoc autoformatting into multiple formatters).  For that:
```suggestion
```

I should maybe also add a formatter for that line of hyphens to make sure it's the same length as the section name.